flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitor.java

```diff
@@ -234,7 +235,9 @@ public WebRuntimeMonitor(
 			.GET("/jobs/:jobid/checkpoints", handler(new JobCheckpointsHandler(currentGraphs)))
 
 			.GET("/taskmanagers", handler(new TaskManagersHandler(DEFAULT_REQUEST_TIMEOUT)))
-			.GET("/taskmanagers/:" + TaskManagersHandler.TASK_MANAGER_ID_KEY, handler(new TaskManagersHandler(DEFAULT_REQUEST_TIMEOUT)))
+			.GET("/taskmanagers/:" + TaskManagersHandler.TASK_MANAGER_ID_KEY + "/metrics", handler(new TaskManagersHandler(DEFAULT_REQUEST_TIMEOUT)))
+			.GET("/taskmanagers/:" + TaskManagersHandler.TASK_MANAGER_ID_KEY + "/log", new TaskManagerLogHandler(retriever, actorSystem.dispatcher(), jobManagerAddressPromise.future(), timeout, true))
```

Comment:
I think we shouldn't use the dispatcher of the `ActorSystem` as the `ExecutorService` here. The `ActorSystem` is usually the `ActorSystem` of the JM and we shouldn't put load on its `ThreadPool`. I would rather use a dedicated `ThreadPool` for the execution of the futures.